cfme/tests/automate/custom_button/test_generic_objects.py

```diff
@@ -345,4 +361,39 @@ def test_custom_button_open_url_evm_obj(setup_obj, button_group):
     Bugzilla:
         1550002
     """
-    pass
+    group, obj_type = button_group
+
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(start="btn", separator="_"),
+        hover=fauxfactory.gen_alphanumeric(),
+        open_url=True,
+        system="Request",
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+
+    # TODO: Move windows handling functionality to browser
+    initial_count = len(view.browser.selenium.window_handles)
+    main_window = view.browser.selenium.current_window_handle
+    custom_button_group.item_select(button.text)
+
+    wait_for(
+        lambda: len(view.browser.selenium.window_handles) > initial_count,
+        timeout=120,
```

Comment:
2 min to wait for a tab to open seems a bit long -- can this be shortened at all?

Reply:
Its maximum time... as we are using it in `wait_for`; If condition satisfy it will take like `10 sec`. can i reduce it to 60s?

Reply:
60s would be better, my only concern with using large timeouts is in the case of a failure. Why wait 2 min when you know after 30 sec that condition has failed? This can potentially add time to our test runs.

Reply:
ack :). after all timeout is decided by try and error...